packages/form-js-editor/src/core/FieldFactory.js

```diff
@@ -0,0 +1,61 @@
+import Ids from 'ids';
+
+
+export default class FieldFactory {
+
+  /**
+   * @constructor
+   *
+   * @param { import('@bpmn-io/form-js-viewer').FormFields } formFields
+   * @param { import('./EventBus').default } eventBus
+   */
+  constructor(formFields, eventBus) {
+    this._ids = new Ids([ 32, 36, 1 ]);
+
+    this._formFields = formFields;
+
+    eventBus.on('diagram.clear', () => {
+      this._ids.clear();
+    });
+  }
+
+  create(attrs) {
+
+    const {
+      type
+    } = attrs;
+
+    if (!this._formFields.get(type)) {
+      throw new Error(`form field of type <${ type }> not supported`);
+    }
+
+    const field = {
+      ...attrs
+    };
+
+    this._ensureId(field);
+
+    return field;
+  }
+
+  _ensureId(field) {
+
+    if (field.id) {
```

Comment:
When creating a new field this is always true since we still assign an ID when calling `create` (e.g. https://github.com/bpmn-io/form-js/commit/4573d06ba8714db25599a96fbf9948c3380bd138#diff-eee06db29752cbe4f5885e865499adc2c62b536aaa220ba8a9656572745f17cfR184). We should not assign IDs in `create` anymore.

Reply:
Good point. Totally missed that we handled that thing in each component. Not where I'd expect it, at all :+1:

Reply:
Cf. https://github.com/bpmn-io/form-js/pull/137/commits/bdad7608752d574ba5678ce0a5a47e828afbc6dc and https://github.com/bpmn-io/form-js/pull/137/commits/bd65741834b110a0506a68c45519e9452b0eb432.